chromadb/segment/impl/distributed/segment_directory.py

```diff
@@ -212,8 +216,15 @@ def stop(self) -> None:
 
     @override
     def get_segment_endpoint(self, segment: Segment) -> str:
-        # TODO: This should rendezvous hash the segment ID to a worker given the current memberlist
-        return "segment-worker.chroma:50051"
+        if self._curr_memberlist is None or len(self._curr_memberlist) == 0:
+            raise ValueError("Memberlist is not initialized")
+        assignment = assign(segment["id"].hex, self._curr_memberlist, murmur3hasher)
+
+        if self._chroma_server_grpc_port is not None:
```

Comment:
The defaulting should happen in the constructor

Reply:
nice catch